src/shared/utilities/cliUtils.ts

```diff
@@ -234,88 +234,87 @@ function getToolkitLocalCliPath(): string {
 
     getLogger('channel').info(`Installing SSM CLI from ${ssmInstaller} to ${outDir}...`)
     progress.report({ message: msgInstallingLocal })
     switch (process.platform) {
         case 'darwin': {
-            return new Promise<string>(async (resolve, reject) => {
-                try {
-                    const tempPath = path.join(tempDir, 'tmp')
-                    const pkgArgs = ['--expand', 'session-manager-plugin.pkg', tempPath]
-                    const tarArgs = ['-xzf', path.join(tempPath, 'Payload')]
-                    await new ChildProcess('pkgutil', pkgArgs, { spawnOptions: { cwd: tempDir } }).run()
-                    await new ChildProcess('tar', tarArgs, { spawnOptions: { cwd: tempPath } }).run()
-
-                    fs.copySync(path.join(tempPath, 'usr', 'local', 'sessionmanagerplugin'), outDir, {
-                        recursive: true,
-                    })
-
-                    resolve(finalPath)
-                } catch (err) {
-                    reject(new InstallerError((err as Error).message))
+            try {
+                const tempPath = path.join(tempDir, 'tmp')
+                const pkgArgs = ['--expand', 'session-manager-plugin.pkg', tempPath]
+                const tarArgs = ['-xzf', path.join(tempPath, 'Payload')]
+                await new TimedProcess('pkgutil', pkgArgs).run({ spawnOptions: { cwd: tempDir } })
+                await new TimedProcess('tar', tarArgs).run({ spawnOptions: { cwd: tempPath } })
+
+                fs.copySync(path.join(tempPath, 'usr', 'local', 'sessionmanagerplugin'), outDir, {
+                    recursive: true,
+                })
+
+                return finalPath
+            } catch (err) {
+                if (TimeoutError.isCancelled(err)) {
+                    throw err
                 }
-            })
+                throw new InstallerError((err as Error).message)
+            }
         }
         case 'win32': {
-            return new Promise<string>(async (resolve, reject) => {
-                try {
-                    new admZip(ssmInstaller).extractAllTo(tempDir, true)
-                    const secondZip = path.join(tempDir, 'package.zip')
-                    new admZip(secondZip).extractAllTo(outDir, true)
-
-                    resolve(finalPath)
-                } catch (err) {
-                    if (err) {
-                        reject(new InstallerError((err as Error).message))
-                    }
+            try {
+                new admZip(ssmInstaller).extractAllTo(tempDir, true)
+                const secondZip = path.join(tempDir, 'package.zip')
+                new admZip(secondZip).extractAllTo(outDir, true)
+
+                return finalPath
+            } catch (err) {
+                if (TimeoutError.isCancelled(err)) {
+                    throw err
                 }
-            })
+                throw new InstallerError((err as Error).message)
```

Comment:
I acknowledge that this is my own crappy code originally, but can we elevate the `try/catch` to encompass the whole `switch` statement? This block is duplicated from above, and the Linux case doesn't have a `try`/`catch` (I don't know if I had a reason for this...probably didn't).